### DISCUSSION TITLE
How to create a custom variable and pass the same as part of body in a artillery request

### DISCUSSION THREAD

**Discussion by brahadeeswaran** [artilleryio/artillery#2362] [Artillery]
Have to create a variable, which has its own logic for creation and then pass it to a artillery test. Pls help sample code

**Comment by hassy** [comment]
hey @brahadeeswaran 👋 can you tell us a bit more about what you're trying to do? There's a few different ways to do it depending on your use case. For example you can run `function` step in a HTTP scenario to create custom variables with arbitrary logic: https://www.artillery.io/docs/reference/engines/http#function-steps-in-a-flow
